### ISSUE TITLE
helm lint --strict fails even if default value is defined

### ISSUE THREAD

**Issue by vladimirtiukhtin** [helm/helm#4573]
Hello

In the deployments.yaml I have this
```
    spec:
      containers:
        - name: test
          image: "{{ .Values.registry_address }}/{{ .Values.image.name }}:{{ .Values.image.tag | default .Values.environment }}"
```
In the values.yaml
```
---

# Default values for the chart set.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.

registry_address: registry.example.com
revisionHistoryLimit: 3
environment: production
location: local
```
Anyway ```helm lint --strict``` fails with undefined variable
```
==> Linting synthesis
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "test/templates/deployments.yaml": template: test/templates/deployments.yaml:24:99: executing "test/templates/deployments.yaml" at <.Values.test...>: map has no entry for key "tag"

Error: 1 chart(s) linted, 1 chart(s) failed
```
In my opinion if default value for variable is set helm should not treat it as undefined
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:17:28Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:08:19Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
```
baremetal
```

**Comment by bacongobbler** [comment]
The core issue here is that the parent value (`.Values.image`) has not been initialized, therefore the chart is calling `.tag` on a `nil` object. In order to fix this, you need to add the following to your values.yaml:

```
image: {}
```

Your chart should then render as normal.

**Comment by walkafwalka** [comment]
@bacongobbler The document for helm lint --strict is "fail on lint warnings". However, running helm lint does not cause the warning for undefined values.

I think either the documentation is misleading or the undefined value warning should be preceded by [WARN] instead of error because helm lint already fails on errors.

**Comment by asergisvt** [comment]
@bacongobbler it actually fails for me. Solution suggested by @bacongobbler works

**Comment by walkafwalka** [comment]
@asergisvt You see the same error using `helm lint` without `--strict`?

**Comment by InAnimaTe** [comment]
> @asergisvt You are see the same error using `helm lint` without `--strict`?

Nope, but then you miss things. Running without `--strict` really kind of defeats the purpose, especially for a CI/CD pipeline.

**Comment by asergisvt** [comment]
@walkafwalka now I see your point. No. I always do --strict

**Comment by malinoff** [comment]
Coming from ansible, it feels like `{{ .Values.foo.bar.baz.xxx.yyy | default something }}` has to result in to `something` regardless of what intermediate values are `nil`. Why should users of a chart care about the exact structure of values if they don't care about changing the defaults at all?

**Comment by Sceat** [comment]
If we have to set `default` values into `Values.yaml` then `default` become obsolete. There should be only one way and if both work i think `--strict` should not exit if a default value is present. the `--strict` opt is mandatory in a CI pipeline

**Comment by bacongobbler** [comment]
I believe this has been fixed in Helm 3.3.0. Please let us know otherwise. Thanks.

**Comment by vivekthangathurai** [comment]
still, it fails with version 3.4.2
